### DISCUSSION TITLE
Search results show task list in !movetotask and !task bangs

### DISCUSSION THREAD

**Discussion by flightmansam** [minbrowser/min#1978] [Show and tell]
I think the reorganisation of tasks is very powerful. But I have found if I have a lot of tasks opened (not collapsed), it becomes cumbersome to drag tabs into the right task using the task overlay. Good thing there is the **!movetotask** bang to help me!

However I am a forgetful person and often forget the exact names/indexs of my tasks. So I have quickly tied together some search result functionality to **!movetotask** (and also **!task** task switcher, as they have the same call signature/functional behaviour).

I also disabled the transition through the task overlay ([L41-L46](https://github.com/flightmansam/min/blob/d73067696c70533db2d1bca5d6990afdd6bb5776/js/searchbar/tabMoverPlugin.js#L41-L46) in my branch), because I find it cumbersome, but could understand if that was put back if this makes it to the master branch. Sidenote: maybe we could have a preference in the settings "Reduce UI animations". This was also the basis for my [discussion here](https://github.com/minbrowser/min/discussions/1975#discussion-4021908).

https://user-images.githubusercontent.com/34013162/164276030-49188b76-9479-4165-addf-334d79e71d51.mov

You can check out the functionality in my [feature branch fork](https://github.com/flightmansam/min/tree/switch-and-movetotask-with-task-list) of min (beware it is a branch based off many of my own custom commits).

Or the specific files are in this commit below:

https://github.com/flightmansam/min/commit/d73067696c70533db2d1bca5d6990afdd6bb5776

Todo items:
- refactor out the getTaskByNameOrNumber(text) function as it is now duplicated in multiple places (sorry, that's just me being agile!)
- I have found that on a bunch of features I have worked on or looked at, we need a way of getting a formatted task name from the task index (i.e. "Task 1" from the 0th element in the tasks TaskList), once again there is a bunch of code duplication around the place (including here). We might want to refactor this into a task (or TaskList) specific functionality.

**Comment by PalmerAL** [comment]
Yeah this is a good idea, do you want to make a PR?

getTaskByNameOrNumber and the method for generating a task name would probably be best as methods in the task interface: https://github.com/minbrowser/min/blob/master/js/tabState/task.js

I agree the animations don't really work well; it's probably fine to just get rid of them.

(Moved to the other discussion) <strike>I'm not convinced by the task name; it looks kind of weird to have it placed there. Maybe if the task name and the menu button had a rounded rect around them and a light background, so that they were all one button, it would look better? But even then, it seems like it's adding more clutter to the navbar (and a lot of people don't use tasks at all, so just having a permanent "task n" sitting in the corner is going to be annoying).</strike>
